### DISCUSSION TITLE
Is there an example of using OpenGL within a Windows App? Perhaps similar to OpenTK.GlWpfControl.

### DISCUSSION THREAD

**Discussion by ToolmakerSteve** [microsoft/WindowsAppSDK#4409] [Q&A]
Migrating a cross-platform OpenGL app to .Net 8.
On Windows, we targeted Xamarin.WPF.  
All we need is to get the client area of a window. To render OpenGL to.

How can we do this on Windows under .Net 8?  

Ideally without dependency on Maui, but would consider that. The deprecation of [OpenGLView in Maui](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.openglview?view=net-maui-7.0) does not give me confidence in that approach.

**Comment by castorix** [comment]
You can use _CreateWindowEx_ (like I did in some samples such [ContainerPanel.xaml.cs](https://github.com/castorix/WinUI3_DesktopDuplication/blob/master/ContainerPanel.xaml.cs) to host Win32 controls
I had tested with OpenGL (_wglCreateContext_ and so on) but it was slower than same code in WPF/Winforms or with Direct2D (maybe I had missed something...))

**Comment by anose001** [comment]
Interested in different options here as well

**Reply by castorix** [reply]
> Interested in different options here as well

There is _WGL_NV_DX_interop2_, which maybe could be used to draw in a SwapChainPanel, but it does not work on my old PC with Intel HD Graphics adapter (_wglDXRegisterObjectNV_ fails with ERROR_OPEN_FAILED)
Only a classic _CreateWindowEx_ works for me (I can get same performance as in Winforms or C++/Win32 by changing some parameters)
